pkg/libdevfile/libdevfile.go

```diff
@@ -69,6 +87,214 @@ func getDefaultCommand(devfileObj parser.DevfileObj, kind v1alpha2.CommandGroupK
 	return groupCmds[0], nil
 }
 
+// ValidateAndGetPushCommands validates the build and the run command,
+// if provided through odo push or else checks the devfile for devBuild and devRun.
+// It returns the build and run commands if its validated successfully, error otherwise.
+func ValidateAndGetPushCommands(
```

Comment:
Do you think we can use `ValidateCommands` function from the devfile library instead of defining separate functions for every command type?
https://github.com/redhat-developer/odo/blob/aa2d935e7f4e7da2bf7249458a5abb9baca2605e/vendor/github.com/devfile/api/v2/pkg/validation/commands.go#L15

Reply:
When looking at the call tree, this `ValidateCommands` seems to be called indirectly when we call `parseDevfile` in the `pkg/devfile` package. So not sure about this, but I intend to have a closer look at simplifying this `libdevfile.go` file when working on either #5775 or #5776.